include/rocksdb/env.h

```diff
@@ -383,6 +383,10 @@ class Env {
   virtual void SetBackgroundThreads(int number, Priority pri = LOW) = 0;
   virtual int GetBackgroundThreads(Priority pri = LOW) = 0;
 
+  virtual Status SetAllowNonOwnerAccess(bool allow_non_owner_access) {
```

Comment:
btw, we also have an `EnvWrapper` class which simply delegates function calls. We should implement it there too.